daft/expressions/expressions.py

```diff
@@ -660,6 +660,33 @@ def is_in(self, other: Any) -> Expression:
         expr = self._expr.is_in(other._expr)
         return Expression._from_pyexpr(expr)
 
+    def between(self, lower: Any, upper: Any) -> Expression:
+        """Checks if values in the Expression are between lower and upper, inclusive.
+
+        Example:
+            >>> # [1, 2, 3, 4] -> [True, True, False, False]
+            >>> col("x").between(1, 2)
+
+        Returns:
+            Expression: Boolean Expression indicating whether values are between lower and upper, inclusive.
+        """
+
+        def cast_to_series_if_needed(item):
+            if not isinstance(item, Expression):
+                if not (
+                    isinstance(item, int) or isinstance(item, float) or isinstance(item, datetime) or (item is None)
+                ):
+                    item = item_to_series("item", item)
+            return item
+
+        lower = cast_to_series_if_needed(lower)
+        upper = cast_to_series_if_needed(upper)
```

Comment:
The inputs don't need to be casted to series, this is only needed when the input is expected to be a list, such as for the `is_in` expression.